Change default agent health check port to avoid conflicts

tklauser opened the pull request:
The default value 9876 for the agent health port introduced in commit
efffbdbebdbf ("daemon: expose HTTP endpoint on localhost for health
checks") conflicts with Istio's ControlZ port. The latter has been in
use for longer. Because the port is only exposed on localhost for use by
liveness/readiness probe, we can change the default value without
breaking users. Thus, change it to 9879 for which there doesn't seem to
be any documented use.

Fixes #19817

tklauser committed:
docs: agent health port is only listening on localhost

Ref. https://github.com/cilium/cilium/blob/dfa6b157e8e9484c65fd938b2b45a4f5f50c61f9/daemon/cmd/agenthealth.go#L25-L31

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
daemon, helm: change default agent health check port to avoid conflicts

The default value 9876 for the agent health port introduced in commit
efffbdbebdbf ("daemon: expose HTTP endpoint on localhost for health
checks") conflicts with Istio's ControlZ port. The latter has been in
use for longer. Because the port is only exposed on localhost for use by
liveness/readiness probe, we can change the default value without
breaking users. Thus, change it to 9879 for which there doesn't seem to
be any documented use.

Suggested-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

gandro reviewed (approved):
Looks pretty straightforward. I assume this does not have any upgrade impact, as this is only used on the pod spec, which is always updated together with the cilium-agent image, correct?

tklauser committed:
ci: checkout correct ref in l4lb workflow

Otherwise e.g. helm charts are checkout out from master instead of the
PR's ref.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

joestringer reviewed (approved):
I'm not sure how the l4lb piece relates to the rest, but all of the individual changes LGTM.
